From 57ed96592c9166d2839f827a0bdb56494b751865 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:36 -0400
Subject: [fs] nfsd: Remove nfs4_lock_state(): nfsd4_test_stateid/nfsd4_free_stateid

Message-id: <1410898849-22126-153-git-send-email-bfields@redhat.com>
Patchwork-id: 95791
O-Subject: [Patch RHEL7 152/165] nfsd: Remove nfs4_lock_state(): nfsd4_test_stateid/nfsd4_free_stateid
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: e7d5dc19ce9800b86dd9e41ff36cc418e9da1fce
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 3199ab5..712d7e7 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4532,11 +4532,9 @@ nfsd4_test_stateid(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  struct nfsd4_test_stateid_id *stateid;
  struct nfs4_client *cl = cstate->session->se_client;
 
- nfs4_lock_state();
  list_for_each_entry(stateid, &test_stateid->ts_stateid_list, ts_id_list)
   stateid->ts_id_status =
    nfsd4_validate_stateid(cl, &stateid->ts_id_stateid);
- nfs4_unlock_state();
 
  return nfs_ok;
 }
@@ -4552,7 +4550,6 @@ nfsd4_free_stateid(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
  struct nfs4_client *cl = cstate->session->se_client;
  __be32 ret = nfserr_bad_stateid;
 
- nfs4_lock_state();
  spin_lock(&cl->cl_lock);
  s = find_stateid_locked(cl, stateid);
  if (!s)
@@ -4593,7 +4590,6 @@ nfsd4_free_stateid(struct svc_rqst *rqstp, struct nfsd4_compound_state *cstate,
 out_unlock:
  spin_unlock(&cl->cl_lock);
 out:
- nfs4_unlock_state();
  return ret;
 }
 
-- 
1.7.1